storage/local/persistence.go

```diff
@@ -606,7 +608,9 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 			if err != nil {
 				return
 			}
-			w.Write(buf)
+			if _, err = w.Write(buf); err != nil {
```

Comment:
That's a good question. I would have assumed that the logging for that happened in the caller, but it doesn't (https://github.com/prometheus/prometheus/blob/master/storage/local/storage.go#L878). Or otherwise, that it would happen in the deferred function at the top, but it doesn't happen there either (https://github.com/prometheus/prometheus/blob/master/storage/local/persistence.go#L549-L551). Seems like another oversight. How about logging in the deferred function?